### PR TITLE
Feature: OpenMP for Local_Orbital_Charge::cal_dk_k

### PR DESCRIPTION
**Pull request by Alcanderian**

### CODE REVIEW

File: source/src_lcao/record_adj.cpp [hunk 1]
```diff
@@ -243,48 +263,59 @@ void Record_adj::for_grid(const Grid_Technique &gt)
...
-	for(int T1=0; T1<GlobalC::ucell.ntype; ++T1)
+	this->iat2ca = new int[GlobalC::ucell.nat];
+	for(int iat=0; iat<GlobalC::ucell.nat; ++iat)
 	{
-		for(int I1=0; I1<GlobalC::ucell.atoms[T1].na; ++I1)
 		{
-			const int iat = GlobalC::ucell.itia2iat(T1,I1);
 			if(gt.in_this_processor[iat])
 			{
+				iat2ca[iat] = na_proc;
```
Comment (Alcanderian) [inline source/src_lcao/record_adj.cpp:273]: Record `ca` here to remove dependency in OpenMP region.
It mainly helps `cal_dk_k`

Comment (dyzheng) [reply source/src_lcao/record_adj.cpp:273]: What is the meaning of "ca"?

Comment (dyzheng) [reply source/src_lcao/record_adj.cpp:273]: A temporary loop count variable changes to a member of class, you can add some annotation for that.

File: source/src_lcao/record_adj.h [hunk 2]
```diff
@@ -32,6 +32,21 @@ class Record_adj
 	int na_proc;
 	int* na_each;
 
+	//--------------------------------------------
```
Comment (Alcanderian) [inline source/src_lcao/record_adj.h:35]: Here is the usage of iat2ca

Comment (Alcanderian) [reply source/src_lcao/record_adj.h:35]: @dyzheng

File: source/src_lcao/record_adj.h [hunk 3]
```diff
@@ -32,6 +32,21 @@ class Record_adj
...
+	// Meaning:
+	// 1. if iat2ca[iat] > 0, it contains the sparse atom index
+	// 2. if iat2ca[iat] < 0, the sparse atom index of iat does not exist 
+	//
+	// Usage: 
+	// 1. iat2ca[iat] > 0 ? na_each[iat2ca[iat]] : 0
+	// 2. iat2ca[iat] > 0 ? info[iat2ca[iat]] : nullptr
+	//--------------------------------------------
+	int* iat2ca;
+
```
Comment (Alcanderian) [inline source/src_lcao/record_adj.h:49]: Here is the usage of iat2ca @dyzheng

**Review by dyzheng — APPROVED** [review]

**Pull request merged**